spec/numberformat.html

```diff
@@ -457,16 +540,33 @@
       <emu-alg>
         1. Let _nf_ be *this* value.
         1. If Type(_nf_) is not Object, throw a *TypeError* exception.
-        1. If _nf_ does not have an [[initializedNumberFormat]] internal slot, throw a *TypeError* exception.
-        1. If _nf_.[[boundFormat]] is *undefined*, then
+        1. If _nf_ does not have an [[initializedNumberFormat]] internal slot whose value is *true*, throw a *TypeError* exception.
+        1. If the [[boundFormat]] internal slot of _nf_ is *undefined*, then
           1. Let _F_ be a new built-in function object as defined in Number Format Functions (<emu-xref href="#sec-number-format-functions"></emu-xref>).
           1. Let _bf_ be BoundFunctionCreate(_F_, _nf_, « »).
           1. Perform ! DefinePropertyOrThrow(_bf_, `"length"`, PropertyDescriptor {[[Value]]: 1, [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true*}).
-          1. Set _nf_.[[boundFormat]] to bf.
+          1. Set _nf_.[[boundFormat]] to _bf_.
         1. Return _nf_.[[boundFormat]].
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-intl.numberformat.prototype.formattoparts">
+      <h1>Intl.NumberFormat.prototype.formatToParts ([ value ])</h1>
```

Comment:
Making the `value` argument to `Intl.NumberFormat.prototype.formatToParts` optional is consistent with the `DateTimeFormat` version of the function.  But it doesn't seem to me that it should be.

There's a somewhat-understandable default value that can be applied for dates, namely the current date/time.  But there's not really a good default value for numbers.  You could argue for it being zero, but wouldn't it be more sensible for the user to pass `0` in that case?  Or you could do what the current patch seems to do and make the default `NaN` (probably not by deliberate choice?  it's just a consequence of `ToNumber(undefined)` evaluating to it).  But it hardly seems like people will generally want to format `NaN` into parts.  As often as not, `NaN` appearing somewhere means some earlier step in execution went awry, and now things are in la-la land.

I think you should make the `value` argument here non-optional by removing the brackets and eliminating the step where "`value` is not provided".  Just let lack-of-argument be filled in with `undefined` by normal spec semantics.  This won't _stop_ users from failing to provide the argument, and if they don't pass the argument, they'll still get dumb `NaN`-handling.  But it'll make `Intl.NumberFormat().formatToParts.length` have a value consistent with how people are expected to use it, and will overwhelmingly use it.